plugin/dnstap/dnstapio/io.go

```diff
@@ -52,10 +58,18 @@ func (dio *dnstapIO) Connect(endpoint string, socket bool) error {
 	if err != nil {
 		return err
 	}
-	go dio.serve()
+	dio.connected = true
 	return nil
 }
 
+// Connect connects to the dnstop endpoint.
+func (dio *dnstapIO) Connect() {
+	if err := dio.connect(); err != nil {
+		log.Printf("[ERROR] No connection to dnstap endpoint")
```

Comment:
Should just return the error here

Reply:
What about case when dnstap listener is working, we start CoreDNS and in the same time we have network fail that will be resolved quickly (on tcp Dial timeout interval)?